api/src/db.js

```diff
@@ -163,4 +164,37 @@ if (UNIT_TEST_ENV) {
 
     throw new Error(`Error while saving docs: ${errors.join(', ')}`);
   };
+
+  const DEFAULT_SECURITY_STRUCTURE = {
+    names: [],
+    roles: [],
+  };
```

Comment:
I think if you call `addRoleToSecurity` multiple times this default object will mutate because it's never cloned. You could either clone this on access on line 184, or I think a better solution would be to change this constant a getter...

```suggestion
  const getDefaultSecurityStructure = () => {
    return { names: [], roles: [] };
  };
```